src/plots/cartesian/dragbox.js

```diff
@@ -712,7 +712,7 @@ module.exports = function dragBox(gd, plotinfo, x, y, w, h, ns, ew) {
                 // This is specifically directed at scatter traces, applying an inverse
                 // scale to individual points to counteract the scale of the trace
                 // as a whole:
-                .selectAll('.points').selectAll('.point')
+                .select('.scatterlayer').selectAll('.points').selectAll('.point')
```

Comment:
That makes sense to me. It's certainly better, though I can't say from direct knowledge that nothing (questionably) puts stray items in the scatterlayer.